### DISCUSSION TITLE
Find favicon by parsing the HTML source.

### DISCUSSION THREAD

**Discussion by libklein** [keepassxreboot/keepassxc#8527] [Ideas]
The IconDownloader currently tried to find website favicons by either using "https://icons.duckduckgo.com" or by appending "/favicon.ico" to the entry's (resolved) URL and several heuristically determined alternative URLs. This requires several requests to the site and external services, and involves some unnecessarily complicated logic that still misses several corner cases (see https://github.com/keepassxreboot/keepassxc/issues/4125).

I'd propose to add an **optional** feature that bypasses this logic and instead uses the page's source code to find the favicon location. We've already had a discussion regarding this feature 2 years ago (https://github.com/keepassxreboot/keepassxc/pull/5509) that raised some concerns w.r.t. parsing the source code, which is why I'd like to discuss this proposal before filing a PR.

I think we can get away with regex parsing. It may fail in some cases (XML is not a regular language) such that an XML based approach would be more "correct" (we'd of course have to make sure to avoid security issues like external references etc., but this is possible with the Qt framework). Still, i believe that the corner cases where regex is insufficient will be too few (if they even exist in the wild) to be worth the extra effort and potential security risks.

Whats your stance on the issue?

**Comment by droidmonkey** [comment]
I'm fine with a simple regex solution.
